pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1942,6 +1942,52 @@ func TestLogicalReplicationSchemaChanges(t *testing.T) {
 	dbA.Exec(t, "ALTER TABLE tab ADD COLUMN newcol INT NOT NULL DEFAULT 10")
 }
 
+// TestUserDefinedTypes verifies that user-defined types are correctly
+// replicated if the type is defined identically on both sides.
+func TestUserDefinedTypes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			},
+		},
+	}
+
+	server, s, dbA, dbB := setupLogicalTestServer(t, ctx, clusterArgs, 1)
+	defer server.Stopper().Stop(ctx)
+
+	_, cleanupA := s.PGUrl(t, serverutils.DBName("a"))
+	defer cleanupA()
+
+	dbBURL, cleanupB := s.PGUrl(t, serverutils.DBName("b"))
+	defer cleanupB()
+
+	// Create the same user-defined type both tables.
+	dbA.Exec(t, "CREATE TYPE my_enum AS ENUM ('one', 'two', 'three')")
+	dbB.Exec(t, "CREATE TYPE my_enum AS ENUM ('one', 'two', 'three')")
+
+	dbA.Exec(t, "CREATE TABLE data (pk INT PRIMARY KEY, val my_enum DEFAULT 'two')")
+	dbB.Exec(t, "CREATE TABLE data (pk INT PRIMARY KEY, val my_enum DEFAULT 'two')")
+	dbA.Exec(t, fmt.Sprintf("ALTER TABLE data %s", lwwColumnAdd))
+	dbB.Exec(t, fmt.Sprintf("ALTER TABLE data %s", lwwColumnAdd))
+
+	dbB.Exec(t, "INSERT INTO data VALUES (1, 'one')")
+	// Force default expression evaluation.
+	dbB.Exec(t, "INSERT INTO data VALUES (2)")
+
+	var jobAID jobspb.JobID
+	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE data ON $1 INTO TABLE data with skip schema check", dbBURL.String()).Scan(&jobAID)
```

Comment:
should we remove the check from here: https://github.com/cockroachdb/cockroach/blob/9f61a38b670e57873509568f67bb665e50cd1935/pkg/sql/catalog/tabledesc/logical_replication_helpers.go#L163-L168

that way we don't need the `skip schema check`

Reply:
i think we should remove it once we fix this issue https://github.com/cockroachdb/cockroach/issues/132167